pkg/catalogsource/creation.go

```diff
@@ -0,0 +1,99 @@
+package catalogsource
```

Comment:
Why are you putting this in a separate package from your `builders` package? And why is there a separate file for `creation` and `deletion`?